### DISCUSSION TITLE
Troubleshooting omnet 5.4.1

### DISCUSSION THREAD

**Discussion by anoikob** [omnetpp/omnetpp#1203] [Q&A] [answered]
Greetings. I'm new to this simulator, and I want to use inet and NeSTiNg for TSN networking simulation. I found a paper where the omnet version used was 5.4.1 . I have tried and successfully played with aloha and some other basic examples, but only in the latest 6.0.3 version. I've tried 4 versions in total. Two of them , 5.4.1 and 5.5.1 give me this error message and Im unable to solve it or interact with the GUI cause it immediately shuts down after closing the message.

![image](https://github.com/user-attachments/assets/a4561387-7a66-43f8-a0b6-2a97b9a81500)

Any ideas? I have not been able to find a file that contains the log format, and I have not been able to create something to bypass it or anything. This error is persistent through all the examples in both versions of the simulator. My OS is Windows 10 and I have followed the omnet installation manual. Your help will be greatly appreciated.

**Comment by levy** [comment]
When you used the newer omnetpp version it probably saved the logformat setting in ~/.qtenvrc and it causes the older versions to fail.

Search for something like:

logformat=%t %l %C%<: %>

**Reply by anoikob** [reply]
Thanks for the answer! Where exactly should I look? Is that a file in the newer version?

**Comment by anoikob** [comment, marked as answer]
Answer found!
The file that omnet was getting that invalid character from was in C/Users/Username/ .qtenvrc , it never accessed the logFormat in the qtenvrc of the project itself. Deleted the '<' character, eliminated the spaces, and the program now runs as it should.

**Comment by rhornig** [comment]
A completely unrelated note: INET 4.5 does support TSN networking. Why are you using NeSTiNg at all?

**Reply by anoikob** [reply]
Because this is the software I was asked to work with ¯\_(ツ)_/¯

**Reply by rhornig** [reply]
That's definitely a valid reason. Just wanted to point out that it's not necessarily the best solution.

**Reply by anoikob** [reply]
For the record, do you know of any omnet related extensions like nesting, for TSN simulations? Just in case, I am asked to find something else. Do you have any recommendations?

**Reply by torokati44** [reply]
AFAIK, CoRE4INET [^1] [^2] was "the original" TSN project for OMNeT++, then came NeSTiNg, and finally all the relevant functionality has been mainlined into INET itself - to my knowledge, at least.

 [^1]: https://core.informatik.haw-hamburg.de/projects/simulation/
 [^2]: https://github.com/CoRE-RG/CoRE4INET

**Reply by anoikob** [reply]
I see, thanks!

**Comment by avarga** [comment]
Just to make it clear, as it has not been said above: **INET implements _significantly_ more of TSN, and on higher quality, than either predecessors Core4INET or NeSTiNg.** There is no sane reason to use either of Core4INET or NeSTiNg for new projects. TSN in INET is also better documented, see e.g. the [showcases](https://inet.omnetpp.org/docs/showcases/tsn/index.html).
